plugins/network-elements/ovs/src/com/cloud/network/element/OvsElement.java

```diff
@@ -488,50 +494,54 @@ public boolean applyPFRules(final Network network, final List<PortForwardingRule
             return true;
         }
 
-        DataCenterVO dcVO = _dcDao.findById(network.getDataCenterId());
-        NetworkTopology networkTopology = _networkTopologyContext.retrieveNetworkTopology(dcVO);
-
-        return networkTopology.applyFirewallRules(network, rules, routers);
+        final DataCenterVO dcVO = _dcDao.findById(network.getDataCenterId());
+        final NetworkTopology networkTopology = _networkTopologyContext.retrieveNetworkTopology(dcVO);
+        for (final DomainRouterVO domainRouterVO : routers) {
+            result = networkTopology.applyFirewallRules(network, rules, domainRouterVO);
```

Comment:
Yep... I know. That's what I said when you first mentioned it. I will change it now, but we have to revisit many places in the Java side as well, and not only the classes I changed, but the VpcManagerImpl.java, for instance.